microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -148,13 +218,71 @@ def _send_subtest_msg(
         environment: Environment,
         test_name: str,
         test_status: TestStatus,
+        test_message: str = "",
     ) -> None:
         subtest_msg = create_test_result_message(
-            SubTestMessage,
-            test_id,
-            environment,
-            test_name,
-            test_status,
+            SubTestMessage, test_id, environment, test_name, test_status, test_message
         )
 
         notifier.notify(subtest_msg)
+
+    def _list_perf_metrics_tests(self, hypervisor="kvm") -> List[str]:
+
+        self._log.debug("Listing the performance test cases")
+        tests_list = []
+        result = self.run(
+            f"tests --hypervisor {hypervisor} --metrics -- -- --list-tests",
+            timeout=self.TIME_OUT,
+            force_run=True,
+            cwd=self.repo_root,
+            no_info_log=False,
+            shell=True,
+            expected_exit_code=0,
+        )
+
+        stdout = result.stdout
+
+        # Ex. String for below regex : "boot_time_ms" (test_timeout = 2s, test_iterations = 10)
+        regex = '\\"(.*)\\" \\('
+
+        pattern = re.compile(regex)
+        tests_list = pattern.findall(stdout)
+
+        self._log.debug(f"Testcases found : {tests_list}")
+        return tests_list
+
+    def _process_perf_metric_test_result(self, output: str) -> str:
+        cnt = 0
+        for line in output.split("\n"):
+            if line.find("git_human_readable") >= 0:
+                cnt -= 1
+                break
+            cnt += 1
+
+        result = "\n".join([i.strip() for i in output.split("\n")[cnt:-5]])
+        self._log.debug(f"Result from testcase stdout : {result}")
+        return result
+
+    def _create_perf_metric_report(self, testcases_result_list) -> None:
+        if not os.path.exists(self.per_mtr_report_file):
+            os.mkdir(
+                os.path.abspath(self.per_mtr_report_file).replace(
+                    os.path.basename(self.per_mtr_report_file), ""
+                )
+            )
```

Comment:
Now that things are in `log_path`. This code is not required.

Reply:
Done.